foursight_core/routes.py

```diff
@@ -0,0 +1,239 @@
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'view_run/{environ}/{check}/{method}', methods=['GET'])
+    def view_run_route(environ, check, method):
+        """
+        Protected route
+        """
+        req_dict = app.current_request.to_dict()
+        domain, context = app.core.get_domain_and_context(req_dict)
+        query_params = req_dict.get('query_params', {})
+        if app.core.check_authorization(req_dict, environ):
+            if method == 'action':
+                return app.core.view_run_action(environ, check, query_params, context)
+            else:
+                return app.core.view_run_check(environ, check, query_params, context)
+        else:
+            return app.core.forbidden_response(context)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'view/{environ}', methods=['GET'])
+    def view_route(environ):
+        """
+        Non-protected route
+        """
+        req_dict = app.current_request.to_dict()
+        domain, context = app.core.get_domain_and_context(req_dict)
+        return app.core.view_foursight(app.current_request, environ, app.core.check_authorization(req_dict, environ), domain, context)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'view/{environ}/{check}/{uuid}', methods=['GET'])
+    def view_check_route(environ, check, uuid):
+        """
+        Protected route
+        """
+        req_dict = app.current_request.to_dict()
+        domain, context = app.core.get_domain_and_context(req_dict)
+        if app.core.check_authorization(req_dict, environ):
+            return app.core.view_foursight_check(app.current_request, environ, check, uuid, True, domain, context)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'history/{environ}/{check}', methods=['GET'])
+    def history_route(environ, check):
+        """
+        Non-protected route
+        """
+        # get some query params
+        req_dict = app.current_request.to_dict()
+        query_params = req_dict.get('query_params')
+        start = int(query_params.get('start', '0')) if query_params else 0
+        limit = int(query_params.get('limit', '25')) if query_params else 25
+        domain, context = app.core.get_domain_and_context(req_dict)
+        return app.core.view_foursight_history(app.current_request, environ, check, start, limit,
+                                      app.core.check_authorization(req_dict, environ), domain, context)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'checks/{environ}/{check}/{uuid}', methods=['GET'])
+    def get_check_with_uuid_route(environ, check, uuid):
+        """
+        Protected route
+        """
+        if app.core.check_authorization(app.current_request.to_dict(), environ):
+            return app.core.run_get_check(environ, check, uuid)
+        else:
+            return app.core.forbidden_response()
```

Comment:
I feel as if there is a repeated pattern here that could be clearer if you created a helper function. I don't think a context manager can work because it can't return a value in a syntactically pleasing way, but a regular function might. e.g.,
```
def if_authorized(app, environ, action):
    if app.core.check_authorization(app.current_request.to_dict(), environ):
        return action()
    else:
        return app.core.forbidden_response()
```
so that you could write something like:
```
    return if_authorized(app, environ, lambda: app.core.run_get_check(environ, check, uuid))
```
rather than the four lines here. Might be it could be a method on the app, in which case it would look like:
```
    return app.if_authorized(environ, lambda: app.core.run_get_check(environ, check, uuid))
```

Reply:
This is the non-React stuff which I tried to touch minimally.